internal/pkg/deploy/cloudformation/project.go

```diff
@@ -221,6 +222,79 @@ func (cf CloudFormation) AddEnvToProject(project *archer.Project, env *archer.En
 	return nil
 }
 
+var getRegionFromClient = func(client cloudformationiface.CloudFormationAPI) (string, error) {
+	concrete, ok := client.(*cloudformation.CloudFormation)
+	if !ok {
+		return "", errors.New("failed to retrieve the region")
+	}
+	return *concrete.Client.Config.Region, nil
+}
+
+// AddPipelineResourcesToProject adds resources needed to support a pipeline
+// to a project. It conditionally creates resources needed for the pipeline
+// in a AWS region in case there's no existing environment
+// (which would have provisioned these resources) for the project
```

Comment:
dumb question: Why do we need to create any pipeline related resources if there are no environments in the project?

Could we not just error asking the user to first create environments? or am I misunderstanding something.

Reply:
regardless of whether there's any existing environment, there's always a possibility that the region we spin up SSM store & CodePipeline does not contain any environment at all, hence missing the usual resources that come with an environment.

Reply:
> dumb question: Why do we need to create any pipeline related resources if there are no environments in the project?
> 
> Could we not just error asking the user to first create environments? or am I misunderstanding something.

regardless of whether there's any existing environment, there's always a possibility that the region we spin up SSM store & CodePipeline does not contain any environment at all, hence missing the usual resources that come with an environment.

Reply:
Right, sorry I meant why are we trying to create a CodePipeline if there are no environments?

Reply:
The codePipeline is always created in the same region as the SSM store. And it's possible that region does not contain any existing environment, regardless of whether some other region might contain environments or not.

Reply:
Clarified offline it was my misunderstanding of how CodePipeline works :D